instrumentation/opentelemetry-instrumentation-tornado/src/opentelemetry/instrumentation/tornado/client.py

```diff
@@ -105,37 +105,52 @@ def _finish_tracing_callback(
     request_size_histogram,
     response_size_histogram,
 ):
+    response = None
     status_code = None
+    status = None
     description = None
-    exc = future.exception()
-
-    response = future.result()
 
-    if span.is_recording() and exc:
+    exc = future.exception()
+    if exc:
+        description = f"{type(exc).__name__}: {exc}"
         if isinstance(exc, HTTPError):
+            response = exc.response
             status_code = exc.code
-        description = f"{type(exc).__name__}: {exc}"
+            status = Status(
+                status_code=http_status_to_status_code(status_code),
+                description=description,
+            )
+        else:
+            status = Status(
+                status_code=StatusCode.ERROR,
+                description=description,
+            )
```

Comment:
I believe you would need to either `span.record_exception()` manually or reraise the exception and let the [sdk](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L1001) handle it for you.

Reply:
updated, now calling `span.record_exception(exc)` here